docker/Dockerfile-test

```diff
@@ -15,6 +15,6 @@ COPY ./ /src
 
 ARG DOCKER_GIT_CREDENTIALS
 RUN python3 -m pip install --upgrade pip
-RUN python3 -m pip install -r requirements-dev.txt
+RUN python3 -m pip install --no-cache-dir .[dev]
```

Comment:
Nice catch here. This approach could even use its own set of optional dependencies (such as `pip install .[test]`) to avoid installing unnecessary dev tools alongside required testing components. See an example of that in SCOS Actions [here](https://github.com/NTIA/scos-actions/blob/1d2b669bc22c8bafe2bcb0531a36088edd468087/pyproject.toml#L55). 

This would be a pretty insignificant change at the moment, since it would only prevent installing pre-commit and hatchling. But if dev tooling ever grows in the future, this is how we should likely handle it.

Reply:
I'll make a corresponding change in scos-tekrsa soon, I missed this when making the switch away from requirements.txt!

Reply:
Done - separated into test and dev